### DISCUSSION TITLE
RFC: `Divider`

### DISCUSSION THREAD

**Discussion by oliverlloyd** [guardian/source#814] [RFC]
# `Divider`

A component to insert an `hr` tag. A version of this [already exists](https://github.com/guardian/dotcom-rendering/blob/fc60c242813d322611283e395e1a58a16c17cad8/src/web/components/elements/DividerBlockComponent.tsx) in DCR.

## Props

```ts
type Props = {
	size?: 'full' | 'partial'; // Full width or limited to 150px
	spaceAbove?: 'tight' | 'loose'; // margin-top at ${space[6]}px or ${space[12]}px
	text?: 'string'; // Text to centre inside the line (eg.: ---- or ---- )
};
```

### With text
![Screenshot 2021-05-12 at 11 57 06](https://user-images.githubusercontent.com/1336821/117964189-39310400-b319-11eb-9d60-0b428c747041.jpg)
### Full
![Screenshot 2021-05-12 at 11 56 25](https://user-images.githubusercontent.com/1336821/117964191-39c99a80-b319-11eb-93b1-71a63d4fb3c4.jpg)
### Partial
![Screenshot 2021-05-12 at 11 54 58](https://user-images.githubusercontent.com/1336821/117964193-3a623100-b319-11eb-8fde-264bab6a3346.jpg)

**Comment by oliverlloyd** [comment]
[Here](https://github.com/guardian/gateway/pull/531)'s a propsal for a `Divider` component in Identity

**Reply by dom-whooley** [reply]
Adding page url's where centred text divider is featured:

- https://profile.theguardian.com/signin

**Reply by oliverlloyd** [reply]
This is now [a PR here](#912 )
